storage/buckets/set_bucket_cors_configuration.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buckets
+
+// [START storage_cors_configuration]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+// setBucketCORSConfiguration sets a CORS configuration on a bucket.
+func setBucketCORSConfiguration(w io.Writer, bucketName string, maxAge time.Duration, methods, origins, responseHeaders []string) error {
+	// bucketName := "bucket-name"
+	// maxAge := time.Hour
+	// methods := []string{"GET"}
+	// origins := []string{"some-origin.com"}
+	// responseHeaders := []string{"Content-Type"}
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+
+	bucket := client.Bucket(bucketName)
+	bucketAttrsToUpdate := storage.BucketAttrsToUpdate{
```

Comment:
Does it make more sense to grab the existing CORS and append to that rather than overwriting? Thoughts @frankyn ?

Reply:
I leave this to @JesseLovelace, when they're back in the office.

Reply:
The other samples do a full replace, so let's do that for consistency. The general use case for this sample is setting CORS for the first time, so the user would probably want to specify their own configuration